src/plugins/data_views/server/content_management/data_views_storage.ts

```diff
@@ -0,0 +1,140 @@
+
+const savedObjectClientFromRequest = async (ctx: StorageContext) => {
+  if (!ctx.requestHandlerContext) {
+    throw new Error('Storage context.requestHandlerContext missing.');
+  }
+
+  const { savedObjects } = await ctx.requestHandlerContext.core;
+  return savedObjects.client;
+};
+
+export class DataViewsStorage implements ContentStorage {
+  constructor() {}
+
+  async get(ctx: StorageContext, id: string): Promise<DataViewGetOut> {
+    const soClient = await savedObjectClientFromRequest(ctx);
+
+    const {
+      saved_object: savedObject,
+      alias_purpose: aliasPurpose,
+      alias_target_id: aliasTargetId,
+      outcome,
+    } = await soClient.resolve<DataViewAttributes>(DataViewSOType, id);
+
+    return { item: savedObject, meta: { aliasPurpose, aliasTargetId, outcome } };
+  }
+
+  async bulkGet(ctx: StorageContext, ids: string[], options: unknown): Promise<any> {
+    return {};
+  }
+
+  async create(
+    ctx: StorageContext,
+    data: DataViewCreateIn['data'],
+    options: DataViewCreateOptions
+  ): Promise<DataViewCreateOut> {
+    const { references, overwrite, id } = options!;
+
+    const createOptions = {
+      id,
+      overwrite,
+      references,
+    };
+
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const result = await soClient.create(DataViewSOType, data, createOptions);
+    return { item: result };
+  }
+
+  async update(
+    ctx: StorageContext,
+    id: string,
+    data: DataViewUpdateIn['data'],
+    options: DataViewUpdateOptions
+  ): Promise<DataViewUpdateOut> {
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const result = await soClient.update<DataViewUpdateIn['data']>(
+      DataViewSOType,
+      id,
+      data,
+      options
+    );
+    return { item: result };
+  }
+
+  async delete(ctx: StorageContext, id: string): Promise<DataViewDeleteOut> {
+    const soClient = await savedObjectClientFromRequest(ctx);
+    await soClient.delete(DataViewSOType, id);
+    return { success: true };
+  }
+
+  async search(
+    ctx: StorageContext,
+    query: SearchQuery,
+    options: DataViewSearchOptions
+  ): Promise<DataViewSearchOut> {
+    const soClient = await savedObjectClientFromRequest(ctx);
+
+    const { included, excluded } = query.tags ?? {};
+    const hasReference: SavedObjectsFindOptions['hasReference'] = included
+      ? included.map((id) => ({
+          id,
+          type: 'tag',
+        }))
+      : undefined;
+
+    const hasNoReference: SavedObjectsFindOptions['hasNoReference'] = excluded
+      ? excluded.map((id) => ({
+          id,
+          type: 'tag',
+        }))
+      : undefined;
+
+    const soQuery: SavedObjectsFindOptions = {
+      type: DataViewSOType,
+      search: query.text,
+      perPage: query.limit,
+      page: query.cursor ? +query.cursor : undefined,
+      defaultSearchOperator: 'AND',
+      searchFields: options.searchFields || ['title', 'name'],
```

Comment:
You are exposing searching accross any fields on the database from the UI. It is better to limit what fields can be searched on.
In maps I added an `onlyTitle` option to avoid that (https://github.com/elastic/kibana/pull/153304/files#diff-3196f0787529ecacf22f90a6081b048f25eb3cf980b548437da12df2879ccdadR287).